internal/server/resource.go

```diff
@@ -1126,11 +1126,11 @@ func newIPSet(f schema.SchemaSetFunc, ips []net.IP) *schema.Set {
 	return schema.NewSet(f, ss)
 }
 
-func setServerSchema(d *schema.ResourceData, s *hcloud.Server) {
-	util.SetSchemaFromAttributes(d, getServerAttributes(d, s))
+func setServerSchema(d *schema.ResourceData, s *hcloud.Server, data_source bool) {
+	util.SetSchemaFromAttributes(d, getServerAttributes(d, s, data_source))
 }
 
-func getServerAttributes(d *schema.ResourceData, s *hcloud.Server) map[string]interface{} {
+func getServerAttributes(d *schema.ResourceData, s *hcloud.Server, data_source bool) map[string]interface{} {
```

Comment:
Could you rename this to `forceSetNetworkAttribute` instead of `data_source` in this method and in `setServerSchema`?